workers/unity/Assets/Playground/Scripts/Cubes/ProcessColorChangeSystem.cs

```diff
@@ -48,24 +48,24 @@ protected override void OnUpdate()
         {
             for (var i = 0; i < data.Length; i++)
             {
-                var component = data.EventUpdate[i];
+                var colorChangeEvents = data.EventUpdate[i];
                 var renderer = data.Renderers[i];
-                foreach (var colorEvent in component.Events)
-                {
-                    renderer.SetPropertyBlock(materialPropertyBlocks[colorEvent.Color]);
-                }
+
+                var lastEvent = colorChangeEvents.Events[colorChangeEvents.Events.Count - 1];
+                renderer.SetPropertyBlock(materialPropertyBlocks[lastEvent.Color]);
             }
         }
 
         private static void PopulateMaterialPropertyBlockMap(
             out Dictionary<Generated.Playground.Color, MaterialPropertyBlock> materialpropertyBlocks)
         {
-            materialpropertyBlocks = new Dictionary<Generated.Playground.Color, MaterialPropertyBlock>();
-            foreach (var keyValuePair in ColorMapping)
+            materialpropertyBlocks = new Dictionary<Generated.Playground.Color, MaterialPropertyBlock>(ColorMapping.Count);
```

Comment:
In `TriggerColorChangeSystem` we have `using Color = Generated.Playground.Color`. Maybe use it here as well?